src/back/wgsl/writer.rs

```diff
@@ -173,6 +182,31 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
+    fn get_struct_name(&self, module: &Module, handle: Handle<Type>) -> String {
+        if module.types[handle].name.is_none() {
+            #[allow(clippy::pattern_type_mismatch)]
+            let shader_stage =
+                self.ep_results
+                    .iter()
+                    .find_map(|(stage, ty)| if *ty == handle { Some(stage) } else { None });
```

Comment:
nit: this would be simpler with just `find()`

Reply:
`find()` returns to us `(shader_stage, ty)` here, but we need only `shader_stage`.

Reply:
right, so you'd do `find(...).0` for this